File: cleanup.py
#!/usr/bin/env python3
"""Strip JS, Google preconnects/refs from the saved Meet HTML and point broken
   local image refs at a single placeholder."""
import re
from pathlib import Path

src = Path("/Users/user/mete/Meet - yrt-kczi-csw.html")
html = src.read_text(encoding="utf-8")
orig_len = len(html)

# 1) Remove all <script>...</script> (including attributes, with or without body)
html, n_script = re.subn(r'<script\b[^>]*>.*?</script\s*>', '', html, flags=re.DOTALL | re.IGNORECASE)
# Self-closing or unclosed safety net
html, n_script2 = re.subn(r'<script\b[^>]*/>', '', html, flags=re.IGNORECASE)

# 2) Remove <noscript>...</noscript>
html, n_noscript = re.subn(r'<noscript\b[^>]*>.*?</noscript\s*>', '', html, flags=re.DOTALL | re.IGNORECASE)

# 3) Strip inline event handlers (on*="..." / on*='...')
html, n_on = re.subn(r'''\s+on[a-z]+\s*=\s*"[^"]*"''', '', html, flags=re.IGNORECASE)
html, n_on2 = re.subn(r"""\s+on[a-z]+\s*=\s*'[^']*'""", '', html, flags=re.IGNORECASE)

# 4) Remove <link rel="preconnect|dns-prefetch"> pointing at Google domains
def _drop_link(m):
    return ''
html, n_pre = re.subn(
    r'<link\b[^>]*\brel\s*=\s*"(?:preconnect|dns-prefetch)"[^>]*\bhref\s*=\s*"https?://[^"]*(?:google|gstatic)[^"]*"[^>]*>',
    _drop_link, html, flags=re.IGNORECASE)

# 5) Replace broken ./Meet - yrt-kczi-csw_files/... references with placeholder
html, n_local = re.subn(r'\./Meet - yrt-kczi-csw_files/[^"\'<>)\s]+',
                        'assets/img/placeholder.svg', html)

# 6) Rename Google → App in identifiers/font names (we've already stripped Google URLs/JS)
html, n_G = re.subn(r'Google', 'App', html)
html, n_g = re.subn(r'google', 'app', html)

# 7) Tidy stray empty <head> whitespace
html = re.sub(r'\n{3,}', '\n\n', html)

src.write_text(html, encoding="utf-8")

print(f"scripts removed:        {n_script + n_script2}")
print(f"noscript removed:       {n_noscript}")
print(f"inline handlers:        {n_on + n_on2}")
print(f"google preconnects:     {n_pre}")
print(f"local img refs -> ph:   {n_local}")
print(f"Google -> App:          {n_G}")
print(f"google -> app:          {n_g}")
print(f"size: {orig_len} -> {len(html)} bytes  ({100*(orig_len-len(html))/orig_len:.1f}% smaller)")

File: cleanup.php
<?php
/**
 * Прибирає JS, Google-preconnect-и/посилання зі збереженої сторінки Meet і
 * вказує биті локальні посилання на один placeholder. (Порт cleanup.py.)
 *
 *     php cleanup.php
 */
$src = '/Users/user/mete/Meet - yrt-kczi-csw.html';
if (!is_file($src)) {
    fwrite(STDERR, "Не знайдено: $src\n");
    exit(1);
}
$html = file_get_contents($src);
$origLen = strlen($html);

// 1) Прибрати всі <script>...</script>
$html = preg_replace('#<script\b[^>]*>.*?</script\s*>#is', '', $html, -1, $nScript);
$html = preg_replace('#<script\b[^>]*/>#i', '', $html, -1, $nScript2);
// 2) Прибрати <noscript>...</noscript>
$html = preg_replace('#<noscript\b[^>]*>.*?</noscript\s*>#is', '', $html, -1, $nNoscript);
// 3) Inline-обробники on*="..." / on*='...'
$html = preg_replace('#\s+on[a-z]+\s*=\s*"[^"]*"#i', '', $html, -1, $nOn);
$html = preg_replace("#\s+on[a-z]+\s*=\s*'[^']*'#i", '', $html, -1, $nOn2);
// 4) <link rel="preconnect|dns-prefetch"> на Google-домени
$html = preg_replace(
    '#<link\b[^>]*\brel\s*=\s*"(?:preconnect|dns-prefetch)"[^>]*\bhref\s*=\s*"https?://[^"]*(?:google|gstatic)[^"]*"[^>]*>#i',
    '', $html, -1, $nPre
);
// 5) Биті ./Meet - yrt-kczi-csw_files/... → placeholder
$html = preg_replace('#\./Meet - yrt-kczi-csw_files/[^"\'<>)\s]+#', 'assets/img/placeholder.svg', $html, -1, $nLocal);
// 6) Google → App, google → app
$html = preg_replace('#Google#', 'App', $html, -1, $nG);
$html = preg_replace('#google#', 'app', $html, -1, $ng);
// 7) Прибрати зайві порожні рядки
$html = preg_replace('#\n{3,}#', "\n\n", $html);

file_put_contents($src, $html);

printf("scripts removed:        %d\n", $nScript + $nScript2);
printf("noscript removed:       %d\n", $nNoscript);
printf("inline handlers:        %d\n", $nOn + $nOn2);
printf("google preconnects:     %d\n", $nPre);
printf("local img refs -> ph:   %d\n", $nLocal);
printf("Google -> App:          %d\n", $nG);
printf("google -> app:          %d\n", $ng);
printf("size: %d -> %d bytes  (%.1f%% smaller)\n", $origLen, strlen($html), 100 * ($origLen - strlen($html)) / $origLen);
